File: MedicalClinicBackend/src/main/java/pl/aticode/medicalclinic/model/VisitData.java
package pl.aticode.medicalclinic.model;

import java.time.LocalDate;
import java.time.LocalTime;

import lombok.Getter;
import lombok.Setter;
import pl.aticode.medicalclinic.entity.visit.VisitStatus;

/**
 * Model using to receive visit data from frontend. 
 * @author aticode.pl
 *
 */
@Getter @Setter
public class VisitData {

	private LocalDate visitDate;
	private LocalTime visitTime;
	private String doctorUsername;
	private String patientUsername;
	private String employeeUsername;
	private String patientAilmentsInfo;
	private LocalDate visitDateFrom;
	private LocalDate visitDateTo;
	private VisitStatus visitStatus;
}

File: MedicalClinicFrontend/src/app/models/visit-data.ts
import { VisitStatus } from '../enums/visit-status.enum';

export class VisitData {


  constructor(private visitDate?: string, private visitTime?: string, private doctorUsername?: string,
              private patientUsername?: string, private employeeUsername?: string, private patientAilmentsInfo?: string,
              private visitDateFrom?: string, private visitDateTo?: string, private visitStatus?: VisitStatus) { }
}
